packages/rum-core/src/performance-monitoring/performance-monitoring.js

```diff
@@ -186,16 +186,24 @@ class PerformanceMonitoring {
   }
 
   filterTransaction(tr) {
-    var performanceMonitoring = this
-    var transactionDurationThreshold = this._configService.get(
+    const transactionDurationThreshold = this._configService.get(
       'transactionDurationThreshold'
     )
-    var duration = tr.duration()
-    if (
-      !duration ||
-      duration > transactionDurationThreshold ||
-      !tr.spans.length
-    ) {
+    const duration = tr.duration()
+    if (!duration || duration > transactionDurationThreshold) {
+      this._logginService.debug(
+        `Transaction was discarded! transaction duration: ${duration} contains falsy value or it is greater than transactionDurationThreshold: ${transactionDurationThreshold} config`
```

Comment:
Nit: "transaction" should be capitalised, since it follows an exclamation mark.

EDIT: ah I see you're being consistent with existing logging. Perhaps defer this and fix at once another time.